docs/src/pages/guides/FHIRPerformanceGuide.md

```diff
@@ -0,0 +1,800 @@
+  FROM Patient_RESOURCES R, 
+       Patient_LOGICAL_RESOURCES LR 
+ WHERE LR.LOGICAL_ID = ? 
+   AND R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID
+```
+
+When the data is currently cached by the database, this query takes less than a millisecond to execute, as shown in the following execution plan analysis from PostgreSQL:
+
+```
+Nested Loop  (cost=0.84..10.88 rows=1 width=1335) (actual time=0.046..0.058 rows=1 loops=1)
+  Buffers: shared hit=8
+  ->  Index Scan using idx_patient_logical_resourceslogical_id on patient_logical_resources lr  (cost=0.42..5.44 rows=1 width=53) (actual time=0.026..0.030 rows=1 loops=1)
+        Index Cond: ((logical_id)::text = '17546b5cd4a-cae29bb0-d6d7-4a1a-a8df-505e2e3a975e'::text)
+        Buffers: shared hit=4
+  ->  Index Scan using patient_resources_prf_in1 on patient_resources r  (cost=0.42..5.44 rows=1 width=1290) (actual time=0.011..0.012 rows=1 loops=1)
+        Index Cond: (resource_id = lr.current_resource_id)
+        Buffers: shared hit=4
+Planning Time: 0.313 ms
+Execution Time: 0.127 ms
+```
+
+## 5.2 Version Read
+
+The FHIR specification supports reading a specific version of a resource:
+
+```
+Patient/17546b5cd4a-cae29bb0-d6d7-4a1a-a8df-505e2e3a975e/_history/1
+```
+
+The resulting query is similar, except in this case, the specified version is requested from the PATIENT_RESOURCES table. Performance is similar to the plain read, depending on what data is currently cached, of course:
+
+```
+SELECT R.RESOURCE_ID, R.LOGICAL_RESOURCE_ID, R.VERSION_ID, 
+       R.LAST_UPDATED, R.IS_DELETED, R.DATA, LR.LOGICAL_ID 
+  FROM Patient_RESOURCES R, Patient_LOGICAL_RESOURCES LR 
+ WHERE LR.LOGICAL_ID = ? 
+   AND R.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID 
+   AND R.VERSION_ID = ?
+```
+
+## 5.3 History
+
+The history query returns all versions of a resource. Because there is no limit to the number of versions for a given resource, the results are ordered by the version_id (resource version number) and paginated using OFFSET and FETCH NEXT ROWS clauses:
+
+```
+  SELECT R.RESOURCE_ID, R.LOGICAL_RESOURCE_ID, R.VERSION_ID, 
+         R.LAST_UPDATED, R.IS_DELETED, R.DATA, LR.LOGICAL_ID 
+    FROM Patient_RESOURCES R, Patient_LOGICAL_RESOURCES LR 
+   WHERE LR.LOGICAL_ID = ? 
+     AND R.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID 
+ORDER BY R.VERSION_ID DESC 
+  OFFSET ? ROWS 
+  FETCH NEXT ? ROWS ONLY 
+```
+
+In order to include an overall count of the number of resource versions, the IBM FHIR Server must execute an additional SQL query:
+
+```
+SELECT COUNT(R.VERSION_ID) 
+  FROM Patient_RESOURCES R, 
+       Patient_LOGICAL_RESOURCES LR 
+ WHERE LR.LOGICAL_ID = ? 
+   AND R.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID;
+```
+
+In most cases the history queries will execute very quickly. Performance will be slower for cases where a single resource has thousands of versions. To avoid this, ingestion pipelines must ensure they only update a version when necessary.
+
+## 5.4 Search Examples
```

Comment:
I'm not sure if this really belongs here, but its nice to have it somewhere.

Reply:
Are you thinking to put all performance-related search tips under here somewhere, or is there a different spot for those.
I'm thinking things like:
* use `:exact` modifier for string searches
* use `_summary` or `_elements` to reduce the payload size
* etc

Reply:
Any tips we can think of should go here I think